Tests/NIOCoreTests/AsyncChannel/AsyncChannelTests.swift

```diff
@@ -41,7 +41,7 @@ final class AsyncChannelTests: XCTestCase {
         let thirdRead = try await iterator.next()
         XCTAssertNil(thirdRead)
 
-        try await channel.close()
+        try await channel.closeFuture.get()
```

Comment:
This change is interesting. It took me a second to figure out what was going on but it looks like the compiler resolves to a different method when SPI was in play and it incorrectly resolved to the default init instead of the one overloaded with `Outbound == Never`. Removing the SPI changed that and the test correctly failed with `Channel already closed`